src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -161,7 +161,7 @@ protected Task<AuthenticationTicket> AuthenticateOnceAsync()
         {
             if (_authenticateTask == null)
             {
-                _authenticateTask = AuthenticateAsync();
+                _authenticateTask = HandleAuthenticateAsync();
```

Comment:
It's odd that you renamed AuthenticateAsync -> HandleAuthenticateAsync, but left AuthenticateOnceAsync as is. HandleAuthenticateOnceAsync?